### PR TITLE
Add a TODO to enable admission control in federation apiserver when the support is added.

### PR DESCRIPTION
**Pull request by madhusudancs**
Only the last commit here needs review.

Depends on #26951.

cc @kubernetes/sig-cluster-federation

[Analytics]()

### CODE REVIEW

File: federation/cluster/common.sh [hunk 1]
```diff
@@ -24,7 +24,14 @@
...
 : "${KUBE_ROOT?Must set KUBE_ROOT env var}"
 
+# Provides the $KUBERNETES_PROVIDER variable and detect-project function
+source "${KUBE_ROOT}/cluster/kube-util.sh"
+
+# Populates $PROJECT
+detect-project
+
 FEDERATION_IMAGE_REPO_BASE=${FEDERATION_IMAGE_REPO_BASE:-'gcr.io/google_containers'}
+FEDERATION_PUSH_REPO_BASE=${FEDERATION_PUSH_REPO_BASE:-gcr.io/${PROJECT}}
```
Comment (quinton-hoole) [inline federation/cluster/common.sh]: Does this work for projects of the form example.com:projectname ?
e.g. in my case my project is google.com:fooproject, and my push repo base is:
gcr.io/google.com/fooproject

Comment (quinton-hoole) [inline federation/cluster/common.sh]: I checked the code for detect-project, and it doesn't look like it will work for the above case.

Comment (madhusudancs) [inline federation/cluster/common.sh]: Yeah, this doesn't support domain-scoped project names. I think it is fair to expect people to set `$FEDERATION_PUSH_REPO_BASE` if the project name isn't a plain string. This is a convenience for people who do have plain string project names.

**Comment by nikhiljindal** [comment]
last commit LGTM.
Feel free to self apply the label when depending PRs have merged and you have rebased

File: federation/cluster/common.sh (hunk 1 above)
Comment (quinton-hoole) [inline federation/cluster/common.sh]: Fair enough.  Best to fail explicitly in that case (e.g. detect whether there is a ':' in what comes back from detect-project, and print an error message telling the user what to do).

Comment (madhusudancs) [inline federation/cluster/common.sh]: Done.

**Comment by madhusudancs** [comment]
@nikhiljindal Thanks! Rebased the PR to only have this commit. Adding LGTM label.

**Pull request merged**